docs/source/index.rst

```diff
@@ -33,7 +33,7 @@ SkyPilot cuts your cloud costs:
 
 * :ref:`Managed Spot <Managed Spot Jobs>`: **3x cost savings** using spot VMs, with auto-recovery from preemptions
 * :ref:`Autostop <Auto-stopping>`: hands-free cleanup of idle clusters
-* :ref:`Benchmark <Benchmark>`: find best VM types for your jobs
+* :ref:`Benchmark <Benchmark: Find the Best Hardware for Your Jobs>`: find best VM types for your jobs
```

Comment:
cc @WoosukKwon - I tried to add this subtitle because "Benchmark" itself may get misunderstood as the benchmark results of the system itself.

Reply:
Maybe we should rename this feature ;)